File: kakao/2021/kakao2021_4.py
def solution(n, s, a, b, fares):
    ret=1<<32
    adj=[[1<<32]*(n+1)for _ in range(n+1)]
    for i in fares:adj[i[0]][i[1]]=i[2];adj[i[1]][i[0]]=i[2]
    for i in range(n+1):adj[i][i]=0
    for k in range(n+1):
        for i in range(n+1):
            for j in range(n+1):
                if adj[i][j]>adj[i][k]+adj[k][j]:adj[i][j]=adj[i][k]+adj[k][j]
    for i in range(n+1):ret=min(ret,adj[s][i]+adj[i][a]+adj[i][b])
    return ret

File: kakao/2021/kakao2021_4.java
class Solution {
  public int solution(int n, int s, int a, int b, int[][] fares) {
      int answer = Integer.MAX_VALUE;
      int[][] adj = new int[n+1][n+1];
      for(int i=0;i<n+1;i++)
      {
          for(int j=0;j<n+1;j++)
          {
              if(i==j)
                  adj[i][j] = 0;
              else
                  adj[i][j] = 1000000;
          }
      }
      for(int i=0;i<fares.length;i++)
      {
          int[] tmp = fares[i];
          adj[tmp[0]][tmp[1]] = tmp[2];
          adj[tmp[1]][tmp[0]] = tmp[2];
      }
      for(int k = 0;k<n+1;k++)
          for(int i=0;i<n+1;i++)
              for(int j=0;j<n+1;j++)
                  if(adj[i][j]>adj[i][k]+adj[k][j])
                      adj[i][j] = adj[i][k]+adj[k][j];
      for(int i=0;i<n+1;i++)
          answer = Math.min(answer,adj[s][i]+adj[i][a]+adj[i][b]);
      return answer;
  }
}